thunder/series/readers.py

```diff
@@ -309,16 +320,22 @@ def frombinary(path, ext='bin', conf='conf.json', dtype=None, shape=None, skip=0
     path = normalize_scheme(path, ext)
 
     from numpy import dtype as dtype_func
-    recordsize = dtype_func(dtype).itemsize * (shape[-1] + skip)
+    nelements = shape[-1] + skip
+    recordsize = dtype_func(dtype).itemsize * nelements
 
     if spark and isinstance(engine, spark):
         lines = engine.binaryRecords(path, recordsize)
-        raw = lines.map(lambda x: frombuffer(buffer(x, 0, recordsize), dtype=dtype)[skip:])
-        rdd = raw.zipWithIndex().map(lambda (ary, idx): ((idx,), ary))
+        raw = lines.map(lambda x: frombuffer(buffer(x), offset=0, count=nelements, dtype=dtype)[skip:])
+
+        def switch(record):
+            ary, idx = record
+            return (idx,), ary
+
+        rdd = raw.zipWithIndex().map(switch)
 
         if shape and len(shape) > 2:
             expand = lambda k: unravel_index(k[0], shape[0:-1])
-            rdd = rdd.map(lambda (k, v): (expand(k), v))
+            rdd = rdd.map(lambda kv: (expand(kv[0]), kv[1]))
```

Comment:
My method here was the following -- I considered two cases:

1. When the key-value pair represented the standard Thunder record of `(index, ndarray)` I used the shorter but less interpretable `kv` inside a lambda.

2. When the key-value pair was not the standard Thunder record, I explicitly defined a new function so that the more explicitly argument names would help someone reading the code to understand what the objects being manipulated are.

To me, this makes sense, but I'll switch it if you're not swayed by my argument.